src/module/skill-actions.ts

```diff
@@ -18,46 +14,57 @@ export interface ActorSkillAction {
   visible: boolean;
 }
 
-export default class SkillAction {
+export interface SkillActionData {
   key: string;
-  itemName: string;
   label: string;
   icon: string;
-  modifier: string;
-  enabled: boolean;
-  includeMap: boolean;
+  proficiencyKey: string;
+  trainingRequired: boolean;
+  featRequired: boolean;
   actor: Actor;
+}
+
+export class SkillAction {
+  data: SkillActionData;
+
+  constructor(data: SkillActionData) {
+    data.icon = 'systems/pf2e/icons/spells/' + data.icon + '.webp';
+    this.data = data;
+  }
+
+  get actor() {
+    return this.data.actor;
+  }
+
+  get key() {
+    return this.data.key;
+  }
 
-  constructor(
-    key: string,
-    label: string,
-    proficiencyKey: string,
-    trainingRequired: boolean,
-    icon: string,
-    featRequired: boolean,
-    actor: Actor,
-    options: SkillActionOptions = {},
-  ) {
-    const skill = actor.data.data.skills[proficiencyKey];
-
-    this.key = key;
-    this.itemName = label;
-    this.actor = actor;
-    this.label = game.i18n.localize(skill.label) + ': ' + label;
-    this.modifier = (skill.value >= 0 ? ' +' : ' ') + skill.value;
-    this.enabled =
-      (skill._modifiers[1].modifier > 0 && trainingRequired) ||
-      (!trainingRequired && this.hasFeat(label, featRequired));
-    this.icon = 'systems/pf2e/icons/spells/' + icon + '.webp';
-    this.includeMap = options.includeMap;
+  get skill() {
+    return this.actor.data.data.skills[this.data.proficiencyKey];
   }
 
   get visible() {
     return this.actorData?.visible ?? true;
   }
 
-  get actionItem() {
-    return ActionsIndex.instance.get(this.itemName);
+  get pf2eItem() {
+    return ActionsIndex.instance.get(this.data.label);
+  }
+
+  getData({ allVisible }: { allVisible: boolean }) {
+    const enabled =
+      (!this.data.trainingRequired || this.skill._modifiers[1].modifier > 0) &&
+      (!this.data.featRequired || this.hasFeat()) &&
+      (this.visible || allVisible);
+
+    return {
+      ...this.data,
```

Comment:
Having all data in `this.data` seems convenient for this use case. Maybe that's one of the reasons foundry uses this pattern.